src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java

```diff
@@ -268,6 +274,14 @@ public B withInstanceInitializer(BiConsumer<ClassLoader, Integer> instanceInitia
         return (B) this;
     }
 
+    public B withDataDirCount(int datadirCount)
+    {
+        assert datadirCount > 0;
```

Comment:
would be nice to leave a message, like `"data dir count requires a positive number but given " + datadirCount`

Reply:
fixed